husky_services/util.js

```diff
@@ -314,36 +314,59 @@ define(['sprintf'], function(sprintf) {
     };
 
     /**
-     * Returns a percentage (0 to 1) on how similar two strings are
-     * http://stackoverflow.com/questions/10473745/compare-strings-javascript-return-of-likely
+     * Compares two url, using their individual parts (sections between "/")
+     * The function tries to match each part of the first url to a part
+     * in the second url.
      *
-     * @param x {String} the first string
-     * @param y the {String} second string
-     * @returns {number} the percentage on how similar two strings are
+     * @param {String} matchUrl The url where each part is tried to match
+     * @param {String} toMatchUrl The url against which parts are matched
+     * @returns {number} The difference between the number of matched parts and not matched parts.
      */
-    Util.prototype.compareStrings = function(x, y) {
-        var simpleCompare = function(a, b) {
-            var lengthA = a.length;
-            var lengthB = b.length;
-            if (lengthA === 0 && lengthB === 0) {
-                return 1;
-            }
-            var equivalency = 0;
-            var minLength = (a.length > b.length) ? b.length : a.length;
-            var maxLength = (a.length < b.length) ? b.length : a.length;
-
-            for (var i = 0; i < minLength; i++) {
-                if (a[i] == b[i]) {
-                    equivalency++;
+    Util.prototype.compareURIs = function(matchUrl, toMatchUrl) {
+        var matchParts = matchUrl.split('/'),
+            toMatchParts = toMatchUrl.split('/'),
+            successfulMatches,
+            countNotMatchedParts,
+
+            /**
+             * Calculates the number of matches parts of the first array
+             * have with parts of the second array
+             *
+             * @param {Array} parts1
+             * @param {Array} parts2
+             * @returns {number}
+             */
+            getSuccessfulMatches = function(parts1, parts2) {
```

Comment:
Does this method have to be a closure in another function? This could easily produce memory leaks. Can't you put it outside? You probably can, because this method does not touch anything outside of itself.